### PR TITLE
News downloader wifi handling

### PR DESCRIPTION
**Pull request by mwoz123**
Handling wifi in NewsDownloader. 
Not tested. Testers welcome.

### CODE REVIEW

File: plugins/newsdownloader.koplugin/main.lua [hunk 1]
```diff
@@ -82,6 +83,22 @@ function NewsDownloader:addToMainMenu(menu_items)
                 callback = function() self:removeNewsButKeepFeedConfig() end,
             },
             {
+                text = _("Turn on/off WiFi"),
```
Comment (Frenzie) [inline plugins/newsdownloader.koplugin/main.lua]: I'm not sure what the purpose of this submenu is?

Comment (mwoz123) [reply plugins/newsdownloader.koplugin/main.lua]: It's shortcut. In android top menu is not visible so it's faster to turn it here than go to settings -> fullscreen -> disable -> enable disable Wifi in android top menu -> revert the settings to fullscreen.

Comment (Frenzie) [reply plugins/newsdownloader.koplugin/main.lua]: There's no need for such duplication.

Click `Download news` (or whatever) → wifi prompt

To turn off `Settings` → Network → turn off

The Android top menu doesn't come into play.

Comment (mwoz123) [reply plugins/newsdownloader.koplugin/main.lua]: Anyway I guess >90% use cases for News downloader you would like to turn on wifi before operation and turn it off after. 

There could be a prompt before downloading (contained in this code change) [ 100% users need that - as it won't work without internet] 
and after (I thought some users might be annoyed by that, that's why decided to do a separte menu)

Comment (mwoz123) [reply plugins/newsdownloader.koplugin/main.lua]: @Frenzie it's much faster to do it from *current* submenu than going to "Settings"

Comment (mwoz123) [reply plugins/newsdownloader.koplugin/main.lua]: ok. replaced with prompt before and after downloading

File: plugins/newsdownloader.koplugin/main.lua [hunk 2]
```diff
@@ -184,6 +189,7 @@ function NewsDownloader:loadConfigAndProcessFeeds()
             text = T(_("Downloading news finished. Could not process some feeds. Unsupported format in: %1"), unsupported_urls)
         })
     end
+    NetworkMgr:promptWifiOff()
```
Comment (Frenzie) [inline plugins/newsdownloader.koplugin/main.lua]: @poire-z Do you think it'd make sense to add some kind of setting for prompting wifi off after doing something?

Like a `NetworkMgr:promptWifiOffAfterAction()`.

@mwoz123 Better but I still don't like the inconsistency.

Once we have something like #3078 we can reintroduce wifi-info and toggle to the top menu ourselves. (It was there for a little while but it caused the issues pointed out by @Hzj-jie for the network menu, except for *everything*.)

Comment (poire-z) [reply plugins/newsdownloader.koplugin/main.lua]: Why not, as an option (so people who have always wifi on don't get bothered by them).

(But I'm not the best person to ask about that: I had heavily patched my uimanager.lua in 2016 to handle auto-wifi on to my known ssids with hardcoded sh scripts, and auto disconnect after a 2mn timeout - that I had every now and then the need to revert some koreader commits - well, it's ugly, no real way to clean PR that, but it's super practical).

Comment (Frenzie) [reply plugins/newsdownloader.koplugin/main.lua]: That timeout thing makes a lot of sense to me. Personally I just keep it on either until I naturally sleep the device or I artificially trigger a "sleep" as it's faster than going into the menu.

I can see where @mwoz123 is coming from for sure, but parts of the program doing their own thing is not only a maintainability nightmare but also not nice toward users.

So yeah, I'd say skip this line for now and just make it part of your personal patch.

The timeout idea sounds quite appealing I must say.

Comment (mwoz123) [reply plugins/newsdownloader.koplugin/main.lua]: I removed additional submenu (with WiFi on/off) with as you didn't like it and so added this prompt after downloading. 
I want to have control over my reader and have good UX, not have to click around dozens menus to do something (which is btw not so very fast on my Android).

I'd get back to additional submenu with WiFi on/off then. ok?
This way it won't bother users who wants to have wifi enabled all the time. 
(not even sure why someone would like to on Eink reader? On my reader Wifi Drains battery very fast - so I'd be angry if I'd to wait couple minutes for auto-disable wifi).

Comment (Frenzie) [reply plugins/newsdownloader.koplugin/main.lua]: What I personally like or dislike is besides the point. I'm speaking here as one of the KOReader maintainers, trying to keep a consistent user experience with as few future maintenance headaches as possible.

As a user I think it's a great idea. As a maintainer I suppose I could live with a temporary placeholder function (which this isn't), but I'd rather keep the whole thing for a different PR.

Git has branches. They're amazing. Use them. :-)

Comment (mwoz123) [reply plugins/newsdownloader.koplugin/main.lua]: What you mean by "temporary placeholder function" ? Wrapped in a separate function?

Let's continue discussion in  #3395.
I extracted it, so at least wifi on prompt can go now.

**Comment by mwoz123** [comment]
Thanks

**Pull request merged**